### PR TITLE
Added some `min_const_generics` revisions into `const_generics` tests

### PR DESCRIPTION
**Pull request by Amjad50**
Help in #75279. still a lot more to cover though

r? @lcnr

### CODE REVIEW

File: src/test/ui/const-generics/argument_order.rs [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 struct Bad<const N: usize, T> {
+    //[min]~^ ERROR type parameters must be declared prior to const parameters
     arr: [u8; { N }],
     another: T,
 }
 
 struct AlsoBad<const N: usize, 'a, T, 'b, const M: usize, U> {
-    //~^ ERROR lifetime parameters must be declared prior
+    //[full]~^ ERROR lifetime parameters must be declared prior
+    //[min]~^^ ERROR lifetime parameters must be declared prior to const parameters
```
Comment (lcnr) [inline src/test/ui/const-generics/argument_order.rs:14]: as `full` and `min` have the same error message, you don't have to duplicate that here iirc
```suggestion
    //~^ ERROR lifetime parameters must be declared prior
```

File: src/test/ui/const-generics/issues/issue-63322-forbid-dyn.rs [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 trait A {}
 struct B;
 impl A for B {}
 
 fn test<const T: &'static dyn A>() {
-    //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]` to be used
+    //[full]~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]` to be used
+    //[min]~^^ ERROR `&'static (dyn A + 'static)` is forbidden as the type of
+    // a const generic parameter
+    //[min]~| ERROR must be annotated with `#[derive(PartialEq, Eq)]` to be used
```
Comment (lcnr) [inline src/test/ui/const-generics/issues/issue-63322-forbid-dyn.rs:14]:
```suggestion
    //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]` to be used
    //[min]~^^ ERROR `&'static (dyn A + 'static)` is forbidden
```
dup err msg

I also prefer to keep the expected error messages rather short, so we don't have to edit as much if we change them

File: src/test/ui/const-generics/issues/issue-64519.rs [hunk 3]
```diff
@@ -1,7 +1,8 @@
 // check-pass
-
-#![feature(const_generics)]
-//~^ WARN the feature `const_generics` is incomplete
+// revisions: full min
+#![cfg_attr(full, feature(const_generics))]
+#![cfg_attr(full, allow(incomplete_features))]
```
Comment (lcnr) [inline src/test/ui/const-generics/issues/issue-64519.rs:4]: did you delete the now unused `stderr` file here?

File: src/test/ui/const-generics/issues/issue-71169.rs [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
-#![allow(incomplete_features)]
+// revisions: full min
+#![cfg_attr(full, feature(const_generics))]
+#![cfg_attr(full, allow(incomplete_features))]
+#![cfg_attr(min, feature(min_const_generics))]
 
 fn foo<const LEN: usize, const DATA: [u8; LEN]>() {}
-//~^ ERROR the type of const parameters must not
+//[full]~^ ERROR the type of const parameters must not
+//[min]~^^ ERROR the type of const parameters must not
```
Comment (lcnr) [inline src/test/ui/const-generics/issues/issue-71169.rs:8]: dup err msg

File: src/test/ui/const-generics/issues/issue-71381.rs [hunk 5]
```diff
@@ -11,17 +13,21 @@ unsafe extern "C" fn pass(args: PassArg) {
 
 impl Test {
     pub fn call_me<Args: Sized, const IDX: usize, const FN: unsafe extern "C" fn(Args)>(&self) {
-        //~^ ERROR: using function pointers as const generic parameters is forbidden
-        //~| ERROR: the type of const parameters must not depend on other generic parameters
+        //[full]~^ ERROR: using function pointers as const generic parameters is forbidden
+        //[full]~| ERROR: the type of const parameters must not depend on other generic parameters
+        //[min]~^^^ ERROR: using function pointers as const generic parameters is forbidden
+        //[min]~| ERROR: the type of const parameters must not depend on other generic parameters
```
Comment (lcnr) [inline src/test/ui/const-generics/issues/issue-71381.rs:19]:
```suggestion
        //~^ ERROR: using function pointers as const generic parameters is forbidden
        //~| ERROR: the type of const parameters must not depend on other generic parameters
```
dup err msg

File: src/test/ui/const-generics/issues/issue-71381.rs [hunk 6]
```diff
@@ -11,17 +13,21 @@ unsafe extern "C" fn pass(args: PassArg) {
...
     unsafe extern "C" fn trampiline<
         Args: Sized,
         const IDX: usize,
         const FN: unsafe extern "C" fn(Args),
-        //~^ ERROR: using function pointers as const generic parameters is forbidden
-        //~| ERROR: the type of const parameters must not depend on other generic parameters
+        //[full]~^ ERROR: using function pointers as const generic parameters is forbidden
+        //[full]~| ERROR: the type of const parameters must not depend on other generic parameters
+        //[min]~^^^ ERROR: using function pointers as const generic parameters is forbidden
+        //[min]~| ERROR: the type of const parameters must not depend on other generic parameters
```
Comment (lcnr) [inline src/test/ui/const-generics/issues/issue-71381.rs:30]:
```suggestion
        //~^ ERROR: using function pointers as const generic parameters is forbidden
        //~| ERROR: the type of const parameters must not depend on other generic parameters
```
dup err msg

File: src/test/ui/const-generics/issues/issue-71382.rs [hunk 7]
```diff
@@ -13,7 +15,8 @@ impl Test {
     }
 
     fn test<const FN: fn()>(&self) {
-        //~^ ERROR: using function pointers as const generic parameters is forbidden
+        //[full]~^ ERROR: using function pointers as const generic parameters is forbidden
+        //[min]~^^ ERROR: using function pointers as const generic parameters is forbidden
```
Comment (lcnr) [inline src/test/ui/const-generics/issues/issue-71382.rs:19]:
```suggestion
        //~^ ERROR: using function pointers as const generic parameters is forbidden
```
dup err msg

File: src/test/ui/const-generics/issues/issue-71611.rs [hunk 8]
```diff
@@ -1,9 +1,13 @@
...
+#![cfg_attr(full, allow(incomplete_features))]
+#![cfg_attr(min, feature(min_const_generics))]
 
 fn func<A, const F: fn(inner: A)>(outer: A) {
-    //~^ ERROR: using function pointers as const generic parameters is forbidden
-    //~| ERROR: the type of const parameters must not depend on other generic parameters
+    //[full]~^ ERROR: using function pointers as const generic parameters is forbidden
+    //[full]~| ERROR: the type of const parameters must not depend on other generic parameters
+    //[min]~^^^ ERROR: using function pointers as const generic parameters is forbidden
+    //[min]~| ERROR: the type of const parameters must not depend on other generic parameters
```
Comment (lcnr) [inline src/test/ui/const-generics/issues/issue-71611.rs:10]:
```suggestion
    //~^ ERROR: using function pointers as const generic parameters is forbidden
    //~| ERROR: the type of const parameters must not depend on other generic parameters
```
dup err msg

File: src/test/ui/const-generics/issues/issue-72352.rs [hunk 9]
```diff
@@ -1,10 +1,13 @@
...
+#![cfg_attr(full, feature(const_generics))]
+#![cfg_attr(full, allow(incomplete_features))]
+#![cfg_attr(min, feature(min_const_generics))]
 
 use std::ffi::{CStr, CString};
 
 unsafe fn unsafely_do_the_thing<const F: fn(&CStr) -> usize>(ptr: *const i8) -> usize {
-    //~^ ERROR: using function pointers as const generic parameters is forbidden
+    //[full]~^ ERROR: using function pointers as const generic parameters is forbidden
+    //[min]~^^ ERROR: using function pointers as const generic parameters is forbidden
```
Comment (lcnr) [inline src/test/ui/const-generics/issues/issue-72352.rs:10]:
```suggestion
    //~^ ERROR: using function pointers as const generic parameters is forbidden
```

File: src/test/ui/const-generics/issues/issue-73508.rs [hunk 10]
```diff
@@ -1,6 +1,10 @@
-#![feature(const_generics)] //~ WARN the feature `const_generics` is incomplete
+// revisions: full min
+#![cfg_attr(full, feature(const_generics))]
+#![cfg_attr(full, allow(incomplete_features))]
+#![cfg_attr(min, feature(min_const_generics))]
 
 pub const fn func_name<const X: *const u32>() {}
-//~^ ERROR using raw pointers
+//[full]~^ ERROR using raw pointers
+//[min]~^^ ERROR using raw pointers as const generic parameters is forbidden
```
Comment (lcnr) [inline src/test/ui/const-generics/issues/issue-73508.rs:8]:
```suggestion
//~^ ERROR using raw pointers
```
dup err msg

File: src/test/ui/const-generics/std/const-generics-range.rs [hunk 11]
```diff
@@ -1,30 +1,40 @@
...
 // `RangeFull` should be usable within const generics:
 struct _RangeFull<const R: std::ops::RangeFull>;
+//[min]~^ ERROR `std::ops::RangeFull` is forbidden as the type of a const generic parameter
 const RANGE_FULL : _RangeFull<{ .. }> = _RangeFull;
 
 // Regression test for #70155
 // `RangeInclusive` should be usable within const generics:
 struct _RangeInclusive<const R: std::ops::RangeInclusive<usize>>;
+//[min]~^ ERROR `std::ops::RangeInclusive<usize>` is forbidden as the type of a const generic
+// parameter
```
Comment (lcnr) [inline src/test/ui/const-generics/std/const-generics-range.rs:26]: Do multiline expected errors work?
```suggestion
//[min]~^ ERROR `std::ops::RangeInclusive<usize>` is forbidden
```

File: src/test/ui/const-generics/std/const-generics-range.rs [hunk 12]
```diff
@@ -1,30 +1,40 @@
...
 
 // `RangeTo` should be usable within const generics:
 struct _RangeTo<const R: std::ops::RangeTo<usize>>;
+//[min]~^ ERROR `std::ops::RangeTo<usize>` is forbidden as the type of a const generic parameter
 const RANGE_TO : _RangeTo<{ .. 1000 }> = _RangeTo;
 
 // `RangeToInclusive` should be usable within const generics:
 struct _RangeToInclusive<const R: std::ops::RangeToInclusive<usize>>;
+//[min]~^ ERROR `std::ops::RangeToInclusive<usize>` is forbidden as the type of a const generic
+// parameter
```
Comment (lcnr) [inline src/test/ui/const-generics/std/const-generics-range.rs:37]:
```suggestion
//[min]~^ ERROR `std::ops::RangeToInclusive<usize>` is forbidden
```
same as above

File: src/test/ui/const-generics/type-dependent/issue-71382.rs [hunk 13]
```diff
@@ -13,7 +15,8 @@ impl Test {
     }
 
     fn test<const FN: fn() -> u8>(&self) -> u8 {
-        //~^ ERROR using function pointers as const generic parameters is forbidden
+        //[full]~^ ERROR using function pointers as const generic parameters is forbidden
+        //[min]~^^ ERROR using function pointers as const generic parameters is forbidden
```
Comment (lcnr) [inline src/test/ui/const-generics/type-dependent/issue-71382.rs:19]:
```suggestion
        //~^ ERROR using function pointers as const generic parameters is forbidden
```
dup err msg

File: src/test/ui/const-generics/type-dependent/type-mismatch.rs [hunk 14]
```diff
@@ -8,5 +10,6 @@ impl R {
 }
 fn main() {
     assert_eq!(R.method::<1u16>(), 1);
-    //~^ ERROR mismatched types
+    //[full]~^ ERROR mismatched types
+    //[min]~^^ ERROR mismatched types
```
Comment (lcnr) [inline src/test/ui/const-generics/type-dependent/type-mismatch.rs:14]:
```suggestion
    //~^ ERROR mismatched types
```
dup err msg

File: src/test/ui/const-generics/types-mismatch-const-args.rs [hunk 15]
```diff
@@ -11,9 +13,11 @@ struct A<'a, T, const X: u32, const Y: u32> {
 
 fn a<'a, 'b>() {
     let _: A<'a, u32, {2u32}, {3u32}> = A::<'a, u32, {4u32}, {3u32}> { data: PhantomData };
-    //~^ ERROR mismatched types
+    //[full]~^ ERROR mismatched types
+    //[min]~^^ ERROR mismatched types
```
Comment (lcnr) [inline src/test/ui/const-generics/types-mismatch-const-args.rs:17]:
```suggestion
    //~^ ERROR mismatched types
```
dup err msg

File: src/test/ui/const-generics/types-mismatch-const-args.rs [hunk 16]
```diff
@@ -11,9 +13,11 @@ struct A<'a, T, const X: u32, const Y: u32> {
...
     let _: A<'a, u16, {2u32}, {3u32}> = A::<'b, u32, {2u32}, {3u32}> { data: PhantomData };
-    //~^ ERROR mismatched types
+    //[full]~^ ERROR mismatched types
+    //[min]~^^ ERROR mismatched types
```
Comment (lcnr) [inline src/test/ui/const-generics/types-mismatch-const-args.rs:20]:
```suggestion
    //~^ ERROR mismatched types
```
dup err msg

**Review by lcnr — APPROVED** [review]
Thanks for the PR :heart: 

I still have some small nits, after that this is ready to land.

File: src/test/ui/const-generics/std/const-generics-range.rs (hunk 11 above)
Comment (Amjad50) [reply src/test/ui/const-generics/std/const-generics-range.rs]: Not sure, removed

**Comment by lcnr** [comment]
:+1: Thanks

@bors r+ rollup

**Pull request merged**